### ISSUE TITLE
IPv4 address not showing in vCenter

### ISSUE THREAD

**Issue by Yncensus** [vmware/open-vm-tools#636]
### Describe the bug

vCenter showed only the IPv6 address of the VM, after disabling IPv6 no address is shown at all

![image](https://user-images.githubusercontent.com/75368755/218431504-5a3e9685-3c99-4b4f-9f14-23247bc05cf4.png)

### Reproduction steps

1. Create new Linux VM (in our case Debian 11)
2. Install open-vm-tools
3. Configure IP address via systemd-networkd
4. (optional) disable IPv6

### Expected behavior

IPv4 address visible in vCenter as configured in the VM and shown by "ip a"

### Additional context

My guess would be that network configuration via systemd-networkd is currently not supported.
This is default in Ubuntu-based Systems and we matched our debian configuration accordingly. While open-vm-tools on Ubuntu shows the configured IP-address in vCenter without any problems, for Debian VMs it does not.

If this is "not a bug", please convert to a feature request, if possible.

**Comment by lemke1458** [comment]
Open-vm-tools gets the networking info out of the OS, so how its configured shouldn't be an issue.  If 'ip a' sees it, so should OVT.  The OS-level gathering starts here https://github.com/vmware/open-vm-tools/blob/master/open-vm-tools/lib/nicInfo/nicInfoPosix.c#L422

Basic questions:

Did you wait 30+ seconds and refresh the UI after adding the IP?  It takes a minute or so to gather and flush the data through.  Is this IP on a NIC that vSphere knows about, eg backed by a virtual NIC in the virtual hardware?

Tools will exclude virtual guest NICs by default based on the interface name, so its possible that's involved here.  See https://github.com/vmware/open-vm-tools/blob/master/open-vm-tools/tools.conf#L232

Are you using the open-vm-tools package from Debian (not built yourself?)  Have you done any modifications to /etc/vmware-tools/tools.conf that might cause it to exclude these IPs?

Is the IP being added a proper externally accessible IP (eg not a loopback IP)?

Seeing IPv6 data suggests that these possibilities probably aren't involved, but we want to be sure.


We'd need a lot of debug output to see where the issue lives.  The first check would be to see if OVT is publishing the data but the UI isn't showing it.  That can be done in a number of ways; the quickest is probably talking to the ESX directly.

1- log into the ESX host
2 - run 'vim-cmd vmsvc/getallvms'  This will dump a list of every VM on the host.  Find the problem VM and note its id, which is the number in the first column.
3 - run 'vim-cmd vmsvc/get.guest <vmid>'.  That will dump everything open-vm-tools is gathering for guest information, including all the networking data.

If the results of step 3 show the VM's NICSs and IPs, then the issue is in the vSphere UI.  If the data isn't there, then OVT isn't gathering it from some reason.  To debug that, we'd need vmware-vmsvc.log with debugging turned up, and 'ip a' output to compare..  We can get you those steps in a later reply, if the above checks point to an OVT issue.

**Comment by Yncensus** [comment]
Thank you for your quick answer, sorry I did not see it sooner.

> Did you wait 30+ seconds and refresh the UI after adding the IP?

Yes.

> Is this IP on a NIC that vSphere knows about?

Yes, used Adapter Type is VMXNET 3.

> Are you using the open-vm-tools package from Debian (not built yourself?)

Yes, output by apt: open-vm-tools/stable,stable-security,now 2:11.2.5-2+deb11u1 amd64 [installed]

> Have you done any modifications to /etc/vmware-tools/tools.conf?

I used the tools.conf provided by the vmware-tools of vCenter itself, following options are set:
vmsvc.handler, vmsvc.maxOldLogFiles, vmsvc.maxLogSize
These should not interfere, but I listed them here regardless.

> Is the IP being added a proper externally accessible IP?

Yes, even in a public range.

I will gather the needed ESX data with our VMware admin and will get back to you this week at the latest.
